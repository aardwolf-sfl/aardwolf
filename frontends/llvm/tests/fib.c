int fib(int n)
{
    int a, b, i, t;
    a = 0;
    b = 1;

    for (i = 1; i < n; i++) {
        t = a + b;
        a = b;
        b = t;
    }

    return b;
}
