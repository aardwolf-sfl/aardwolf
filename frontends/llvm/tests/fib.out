
function: fib

#1 -> #2  ::  defs: %1 / uses:  [@1, 1, 13]  { arg }
#2 -> #3  ::  defs: %2 / uses:  [@1, 4, 7]
#3 -> #4  ::  defs: %3 / uses:  [@1, 5, 7]
#4 -> #5  ::  defs: %4 / uses:  [@1, 7, 12]
#5 -> #6, #10  ::  defs:  / uses: %4, %1 [@1, 7, 5]
#6 -> #7  ::  defs: %5 / uses: %2, %3 [@1, 8, 11]
#7 -> #8  ::  defs: %2 / uses: %3 [@1, 9, 11]
#8 -> #9  ::  defs: %3 / uses: %5 [@1, 10, 11]
#9 -> #5  ::  defs: %4 / uses: %4 [@1, 7, 25]
#10 ->   ::  defs:  / uses: %3 [@1, 13, 5]  { ret }

@1 = tests/fib.c
